metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -118,6 +116,27 @@ protected void initializeStellar() {
     StellarFunctions.initialize(stellarContext);
   }
 
+  private Map<String, Object> getMetadata(Tuple t, boolean readMetadata) {
+    Map<String, Object> ret = new HashMap<>();
+    if(!readMetadata) {
+      return ret;
+    }
+    ret.put(SimpleStormKafkaBuilder.FieldsConfiguration.TOPIC.getFieldName(), t.getString(2));
```

Comment:
The contract is that the value is the first element in the tuple, the key the 2nd element in the tuple and any metadata from the environment passed in after that.